hazelcast-jet-core/src/main/java/com/hazelcast/jet/Processors.java

```diff
@@ -801,11 +802,18 @@ public static ProcessorMetaSupplier writeFile(@Nonnull String directoryName) {
     }
 
     /**
-     * Returns a supplier of {@link CountDistinctP} processors.
+     * Returns a supplier of processors that will calculate the number of
+     * distinct keys they have seen on input. To work properly the vertex has
+     * to be preceded by an {@link Edge#allToOne() all-to-one} and a {@link
+     * Edge#distributed() distributed} edge. It will emit single {@code
```

Comment:
I don't understand this sentence. Does it need two edges, a single edge with both properties, or any of the above? Does "preceded" mean any edge preceding it in topological order?